### DISCUSSION TITLE
How do I stop Shadowsocks from re-adding "/.*/" rule to the PAC configuration file?

### DISCUSSION THREAD

**Discussion by Leopold702** [shadowsocks/shadowsocks-windows#3291] [Q&A]
I want to use Shadowsocks in blacklist mode while in PAC mode, meaning that only traffic for particular domains should go through a proxy while all other connections bypass it.

I figured out, that I can achieve this by removing the `"/.*/"` rule from the PAC-file and adding the domains to which I want to connect through the proxy to the `user-rule.txt`. This seems to work, but the only problem is that every time I'm adding a new rule to `user-rule.txt`, the `"/.*/"` rule gets automatically re-added to the PAC configuration file. I tried disabling the **Secure local PAC** option, but this didn't help.

Does anyone know, if there is a way to stop this behavior?

**Comment by lolmaus** [comment]
@Leopold702  Have you figured this out?

**Reply by Leopold702** [reply]
I've given up on the Shadowsocks' PAC. What I'm doing instead is using the Proxy SwitchyOmega browser extension to manage the proxying rules while Shadowsocks itself serves as just a gateway.

I've configured it this way:

1. Shadowsocks > System proxy is set to **Disable**
2. In the SwitchyOmega I added a new **SOCKS5** proxy server profile: server **127.0.0.1**, port **1080**
3. Added auto switch rules for all websites that should connect through the proxy
4. Switched SwitchyOmega into auto switch mode

It might be a little counterintuitive, but even when set to **Disable**, Shadowsocks still tunnels the traffic that is explicitly sent to its port (1080 by default).

This works perfectly, and it's much easier to manage websites this way. New websites can be added in just a few clicks from the extension menu.

**Comment by meecosha** [comment]
So sad that this shit is still relevant after two years:(
I just want to add my own websites and not use the GFW file, what is the problem with that — the implementation should be the easiest stuff ever

**Comment by Leopold702** [comment]
I don't think this app will see much further development, and would suggest to look at Sing-box/Nekobox instead.
